asmcomp/selectgen.ml

```diff
@@ -1437,4 +1549,6 @@ let _ =
   Simplif.is_tail_native_heuristic := is_tail_call
 
 let reset () =
-  current_function_name := ""
+  current_function_name := "";
+  Ident.Tbl.clear phantom_lets;
+  dead_phantom_lets := Ident.Set.empty
```

Comment:
As a safe guard, I would also reset the global tables in `emit_fundecl` above. I'm not convinced `reset` is systematically called between two function declarations.